File: euler4.java
class euler4
  {
    public static void main(String[] args) {
        int palprime = 0;
        for (int i = 999; i >= 100; i--) {
            for (int j = 999; j >= 100; j--) {
                int sum = 0;
                int x = i * j;
                int y = x;
                while (y != 0) {
                    int r = y % 10;
                    sum = sum * 10 + r;
                    y = y / 10;
                }
                if (sum == x && sum > palprime)
                    palprime = x;
            }
        }
        System.out.println(palprime);
    }
}

File: euler4.js
let palprime = 0;

for (let i = 999; i >= 100; i--) {
    for (let j = 999; j >= 100; j--) {
        let sum = 0;
        let x = i * j;
        let y = x;
        while (y !== 0) {
            let r = y % 10;
            sum = sum * 10 + r;
            y = Math.floor(y / 10);
        }
        if (sum === x && sum > palprime)
            palprime = x;
    }
}

console.log(palprime);
